process/block/metablock.go

```diff
@@ -1521,6 +1569,13 @@ func (mp *metaProcessor) ApplyBodyToHeader(hdr data.HeaderHandler, bodyHandler d
 		return nil, process.ErrWrongTypeAssertion
 	}
 
+	if check.IfNil(bodyHandler) {
```

Comment:
You can move this check in the first line and also add a similar check for headerHandler

Reply:
done.